packages/puppeteer-extra-plugin-block-resources/index.js

```diff
@@ -91,14 +92,50 @@ class Plugin extends PuppeteerExtraPlugin {
     return this.opts.blockedTypes
   }
 
+  /**
+   * Get the request interception resolution priority.
+   *
+   * Priority for Cooperative Intercept Mode can be configured either through `opts` or by modifying this property.
+   *
+   * @type {number} - A number for the request interception resolution priority.
+   */
+  get interceptResolutionPriority() {
+    return this.opts.interceptResolutionPriority
+  }
+
   /**
    * @private
    */
   onRequest(request) {
     const type = request.resourceType()
     const shouldBlock = this.blockedTypes.has(type)
-    this.debug('onRequest', { type, shouldBlock })
-    return shouldBlock ? request.abort() : request.continue()
+
+    // Cooperative Intercept Mode only available in puppeteer@11+
+    const [currentResolution, currentPriority] = request.interceptResolution
+      ? request.interceptResolution()
+      : []
+
+    // Avoid using resolution `alreay-handled` with typo
+    // Requests are immediately handled in Legacy Mode
+    const alreadyHandled = currentResolution
+      ? currentResolution.endsWith('handled')
+      : true
+
+    this.debug('onRequest', {
+      type,
+      shouldBlock,
+      currentResolution,
+      currentPriority
+    })
+
+    if (alreadyHandled) return
+
+    return shouldBlock
+      ? request.abort('blockedbyclient', this.interceptResolutionPriority)
+      : request.continue(
+          request.continueRequestOverrides(),
```

Comment:
For backward compatibility, consider `request.continueRequestOverrides ? request:continueRequestOverrides() : undefined`. Same idea with `.abort()` and `.respond()`